nixos/modules/services/backup/borgbackup.nix

```
@@ -58,7 +58,7 @@ let
   '' + optionalString (cfg.prune.keep != { }) ''
     borg prune $extraArgs \
       ${mkKeepArgs cfg} \
-      ${optionalString (cfg.prune.prefix != null) "--prefix ${escapeShellArg cfg.prune.prefix} \\"}
+      ${optionalString (cfg.prune.prefix != null) "--glob-archives ${escapeShellArg "${cfg.prune.prefix}*"} \\"}
```

```suggestion
      ${optionalString (cfg.prune.prefix != null) "--glob-archives ${escapeShellArg "${cfg.prune.prefix}*"}"} \
```

That way we don't fail when `prune.prefix` is `null` and `extraPruneArgs` is set.

---

To be honest, I don't understand what happens at the end of that line, nor why your suggestion improves things. (And isn't prefix == null already handled?)

Are these changes you think need to be done as part of the "change --prefix to --glob-archives" commit or improvements you'd like to have, that can be done as follow up?

Sorry, I don't have brain cycles left... unless this causes regressions I suggest to land as-is, and people can make improvements on top of master?

---

The precise situation where this is a problem is when `prune.prefix == null && extraPruneArgs != ""`.

With the backslash inside the optional string, the generated command looks like this:
```sh
echo borg create \
  
  extra args
```
and running it results in
```
borg create
script: line 3: extra: command not found
```

With the backslash on the outside, the generated command looks like this:
```sh
echo borg create \
  \
  extra args
```
and running it results in
```
borg create extra args
```

This doesn't have anything to do with your change, just an improvement. I'll take responsibility for it and just commit it and merge.